quantum/keymap_extras/keymap_hebrew.h

```diff
@@ -0,0 +1,147 @@
+#define IL_KAF  KC_F    // כ
+#define IL_AYIN KC_G    // ע
+#define IL_YOD  KC_H    // י
+#define IL_HET  KC_J    // ח
+#define IL_LAMD KC_K    // ל
+#define IL_FKAF KC_L    // ך
+#define IL_FPE  KC_SCLN // ף
+#define IL_COMM KC_QUOT // ,
+// Row 4
+#define IL_ZAYN KC_Z    // ז
+#define IL_SMKH KC_X    // ס
+#define IL_BET  KC_C    // ב
+#define IL_HE   KC_V    // ה
+#define IL_NUN  KC_B    // נ
+#define IL_MEM  KC_N    // מ
+#define IL_TSDI KC_M    // צ
+#define IL_TAV  KC_COMM // ת
+#define IL_FTSD KC_DOT  // ץ
+#define IL_DOT  KC_SLSH // .
+
+/* Shifted symbols
+ * ┌───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┐
+ * │ ~ │ ! │ @ │ # │ $ │ % │ ^ │ & │ * │ ) │ ( │ _ │ + │   │   │
+ * ├───┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴───┤
+ * │     │   │   │   │   │   │   │   │   │   │   │ } │ { │     │
+ * ├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┘     │
+ * │      │   │   │   │   │   │   │   │   │   │ : │ " │        │
+ * ├──────┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴────────┤
+ * │        │   │   │   │   │   │   │   │ > │ < │ ? │          │
+ * ├────┬───┴┬──┴─┬─┴───┴───┴───┴───┴───┴──┬┴───┼───┴┬────┬────┤
+ * │    │    │    │                        │    │    │    │    │
+ * └────┴────┴────┴────────────────────────┴────┴────┴────┴────┘
+ */
+// Row 1
+#define IL_TILD S(IL_SCLN) // ~
+#define IL_EXLM S(IL_1)    // !
+#define IL_AT   S(IL_2)    // @
+#define IL_PND  S(IL_3)    // #
+#define IL_DLR  S(IL_4)    // $
+#define IL_PERC S(IL_5)    // %
+#define IL_CIRC S(IL_6)    // ^
+#define IL_AMPR S(IL_7)    // &
+#define IL_ASTR S(IL_8)    // *
+#define IL_RPRN S(IL_9)    // )
+#define IL_LPRN S(IL_0)    // (
+#define IL_UNDS S(IL_MINS) // _
+#define IL_PLUS S(IL_EQL)  // +
+#define IL_PIPE S(IL_BSLS) // |
+// Row 2
+#define IL_RCBR S(IL_RBRC) // }
+#define IL_LCBR S(IL_LBRC) // {
+// Row 3
+#define IL_COLN S(IL_FPE)  // :
+#define IL_DQUO S(IL_COMM) // "
+// Row 4
+#define IL_RABK S(IL_TAV)  // >
+#define IL_LABK S(IL_FTSD) // <
+#define IL_QUES S(IL_DOT)  // ?
+
+/* AltGr symbols
+ * ┌───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───┬───────┐
+ * │   │   │   │ € │ ₪ │   │   │   │   │   │   │   │   │       │
+ * ├───┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─────┤
+ * │     │   │   │   │   │   │   │ װ │   │   │   │   │   │     │
+ * ├─────┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┴┬──┘     │
+ * │      │   │   │   │   │   │ ױ │ ײ │   │   │   │   │        │
+ * ├────┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴─┬─┴────────┤
+ * │    │   │   │   │   │   │   │   │   │   │   │   │          │
+ * ├────┼───┴┬──┴─┬─┴───┴───┴───┴───┴───┴──┬┴───┼───┴┬────┬────┤
+ * │    │    │    │                        │    │    │    │    │
+ * └────┴────┴────┴────────────────────────┴────┴────┴────┴────┘
+ */
+// Row 1
+#define IL_EURO ALGR(IL_3)    // €
+#define IL_SHKL ALGR(IL_4)    // ₪
+// Row 2
+#define IL_DVAV ALGR(IL_VAV)  // װ
+// Row 3
+#define IL_DYOD ALGR(IL_YOD)  // ײ
+#define IL_VYOD ALGR(IL_HET)  // ױ
```

Comment:
I can't make any sense of why, but when I test this on my system (Windows 8.1), these three characters are not on the keys assigned here.

- `װ` is on AltGr+QWERTY `Y`
- `ײ` is on AltGr+QWERTY `H`
- `ױ` is on AltGr+QWERTY `G`

The assignments here match what's on Wikipedia, so I'm not sure what to do with this knowledge.

Reply:
Same on Windows 10. They should match what the OSK displays IMO.
Also, I see a bunch of other characters (mostly diacritics) that should probably be added too, for completeness.

Reply:
> Same on Windows 10. They should match what the OSK displays IMO.

DVAV, DYOD, and VYOD are principally used when typing Yiddish in Hebrew lettering. You may not be using the same Windows Hebrew keyboard layout (There are multiple options - see below). They work on Mac, Linux and ChromeOS as I present them here. If there is no uniformity in how they work, then perhaps they should be removed (few use them, as they are just shortcuts for letter combinations that are common).

> Also, I see a bunch of other characters (mostly diacritics) that should probably be added too, for completeness.

This was exactly the subject of the comment that you asked me to remove:
```c
/*
 * There is no uniform method for entering Niqqud, so 
 * we do not attempt to provide a mapping here.
*/
```
TL;DR: _Niqqud_ (plural _niqqudot_) is the (transliterated) Hebrew word for diacritics, and is, put succinctly, a can of worms that I don't think QMK should open under any circumstances, especially given that they are almost never used when typing, and there is no agreement on a standard layout. 

Longer explanation:

Background:
- Hebrew text is an [Abjad](https://en.wikipedia.org/wiki/abjad) writing system, where each glyph stands for a consonant, leaving it to readers to infer or otherwise supply the appropriate vowels.
- _Niqqud_ is used (even when handwriting) only in cases where exact pronunciation is deemed critical, e.g. in prayerbooks or modern transcriptions of scripture. Interestingly, ancient Hebrew artifacts, and even newly created Torah scrolls, do not have _niqqudot_.
- When moveable type was the order of the day, printing was already a laborious process, so the addition of diacritics to a text wasn't really a big deal, and many Hebrew texts were printed with _niqqudot_. 
- But typewriters were a different story, and people typing business letters are lazy, so even if the typewriter supported _niqqud_ (many didn't), it generally was not used.

The end result is that Modern Hebrew is almost never written or printed with any use of diacritics. For instance see [the front page of Ha'aretz](https://www.haaretz.co.il/), a leading Israeli newspaper.  Consequently, no good standard has evolved for entering _niqqudot_. The [wikipedia article](https://en.wikipedia.org/wiki/Hebrew_keyboard#Niqqud) has a table showing the most common input methods, but there are far more than the two methods shown there.

So, I propose to ignore the existence of _niqqud_ in `keymap_hebrew.h`. Adding them will only cause confusion.

It should also be pointed out that are at least five different Hebrew keyboard mappings in common use, and I have only implemented the [SII](https://en.wikipedia.org/wiki/Standards_Institute_of_Israel) standard:

1. [Hebrew](http://kbdlayout.info/kbdheb)
2. [Hebrew (SII Standard)](http://kbdlayout.info/kbdhebl3) ← [implemented in `keymap_hebrew.h`]
3. [Hebrew (Phonetic)](https://egarc.ku.edu/installing-hebrew-ku-homophonic-keyboard-windows-7)
4. [Biblical Hebrew (Tiro)](https://www.sbl-site.org/Fonts/BiblicalHebrewTiroManual.pdf)
5. [Biblical Hebrew (SIL)](https://www.sbl-site.org/Fonts/BiblicalHebrewSILManual.pdf)

Reply:
These are the only two choices of keyboard I have when I add the Hebrew language. The SII layout was the default.
![image](https://user-images.githubusercontent.com/4781841/85096535-e13c6b00-b237-11ea-85b5-675a090b9b83.png)
They seem to be identical except for the AltGr symbols.
Hebrew (Standard):
![image](https://user-images.githubusercontent.com/4781841/85096906-14cbc500-b239-11ea-8020-ef7e8c4525ea.png)
Hebrew:
![image](https://user-images.githubusercontent.com/4781841/85096912-1e552d00-b239-11ea-93e1-8a971930242d.png)

The system I've been going with for these headers is to have `keymap_*.h` be the default Windows one for that language, and `keymap_*_osx_[ansi|iso].h` for the macOS ones. So, I think we should at least fix those three characters, and include the div/mul symbols as well as anything else here that is not specifically Niqqud.

Reply:
I am glad that Windows now has SII as the default; that wasn't the case the last time I used Windows extensively for Hebrew typing (about 10 years ago).

There is an interesting phenomenon here, which may guide us. The majority of daily Hebrew speakers live in the State of Israel (about 6.5 million); but there are lots of folks in the rest of the world who use Hebrew only for liturgical purposes (about 7.6 million), using English (or some other latin-alphabet language) as their primary language. Also there are many who use Yiddish daily (about 0.6 million). I note, however, that those who use Hebrew for liturgical purposes are less likely to type in Hebrew regularly. For instance, I live in the U.S.A.; English is my primary language, but I can read and write both Modern and Biblical Hebrew, and I'm reasonably conversant. I think I type in Hebrew perhaps once or twice a month, mostly when communicating with friends in Israel (in Modern Hebrew) or doing things related to my synagogue (using Biblical Hebrew).

Given the above, I agree with your suggested approach. Shekel, Euro are already in the map; I will add div, mul and degree.

I worry about the parens and brackets though; your images differ from the behavior that I see. Can you please check behavior when typing on a keyboard? When I test on Mac and ChromeOS, I see them reversed when using Hebrew (an RTL language). Shift-9 is "open paren", and Shift-0 is "close paren", and similar for the bracket / brace keys...

Reply:
Yep, they're reversed when actually typing. Dunno why it's not in the OSK...

Reply:
Oops - looks like there are no keycodes defined in keycode.h for `◦`, `×` or `÷`, so I won't do those.

Reply:
Doesn't matter: https://github.com/qmk/qmk_firmware/blob/master/quantum/keymap_extras/keymap_us_international.h#L198

Reply:
...and it looks like װ ,ױ ,ײ are already correct, so I don't think there is anything more to be done.

Reply:
Fixed IL_DVAV, IL_DYOD and IL_VYOD, which were incorrect after all.

Reply:
> Doesn't matter:
Added ◦, × and ÷.